tests/quantization/test_precision_init.py

```diff
@@ -438,15 +440,34 @@ def test_hawq_manual_configs(manual_config_params, hw_config):
     assert table._rows == bit_stats
 
 
-@pytest.mark.parametrize('method_name', ['_calc_traces', '_filter_configs_by_precision_constraints'])
-def test_hawq_raises_error_if_method_returns_none(mocker, method_name):
+def test_hawq_warns__if_method_returns_none(mocker):
+    config = create_hawq_test_config()
+    config.device = 'cuda'
+    config['quantizer_setup_type'] = 'pattern_based'
+    model = MockModel()
+    config = register_default_init_args(config, mocker.stub(), mocker.stub())
+    mocker.patch('nncf.quantization.algo.QuantizationController._do_range_init')
+    mocker.patch('nncf.quantization.init_precision.HAWQPrecisionInitializer._calc_traces')
+
+    mocked_trace = mocker.patch('nncf.quantization.init_precision.HAWQPrecisionInitializer'
+                                '._filter_configs_by_precision_constraints')
+    mocked_trace.return_value = None
+
+    with pytest.warns(RuntimeWarning):
+        create_compressed_model_and_algo_for_test(model, config)
+
+
+def test_hawq_raises_error_if_method_returns_none(mocker):
```

Comment:
So what is the right behaviour that we test for? `_warns__if_method_returns_none` or `_raises_error_if_method_returns_none`? These are mutually exclusive definitions, aren't they?

Reply:
Fixed.